application/helpers/expressions/em_core_helper.php

```diff
@@ -2120,7 +2120,7 @@ private function RDP_RunFunction($funcNameToken, $params)
                                         if (is_numeric($params[0])) {
                                             $result = $funcName(floatval($params[0]));
                                         } else {
-                                            $result = NAN;
+                                            $result = false;
```

Comment:
Phew. Almost impossible to figure out the effect of this change. :sweat:

Reply:
Yes, BUT : return value must be same if function have 1 or 2 or any parameters … 

See https://github.com/LimeSurvey/LimeSurvey/blob/857b89f0df595790363ea3d429ab32d71d2ffaaf/application/helpers/expressions/em_core_helper.php#L2110-L2125

`atan` return false, but not`atan2` … 

I can reset to NAN but : there are an issue

Reply:
Maybe we can look for a unit-test design next week.

Reply:
If you're OK : i revert just this part (atant2 : NAN to false)

Just keep https://github.com/LimeSurvey/LimeSurvey/pull/1409/files#diff-7f2b63ae2aaab1ffe6caffbaae39751cL2144

Reply:
I'm OK with anything, but we need a unit-test for RDP_RunFunction. Let me glance the code.

Reply:
I add it for `0` return , but clearly : the line `$result = $result ? $result : false;` was ONLY HERE if function have 2 parameters.

And NAN return only for atan2, not for all other math function

Reply:
Want me to set it up? The method needs to be made public, and maybe some state needs to be set before running the function.

Reply:
If only NAN (for atan2) function is an issue, i can revert.